k8s/src/main/scala/io/buoyant/k8s/RouteManager.scala

```diff
@@ -1,35 +1,49 @@
 package io.buoyant.k8s
 
+import com.twitter.conversions.time._
+import com.twitter.finagle.Http
+import com.twitter.finagle.param.Label
+import com.twitter.finagle.tracing.NullTracer
 import com.twitter.util._
 import io.buoyant.k8s.IstioPilotClient.RouteRuleConfig
-import java.util.concurrent.atomic.AtomicReference
-import istio.proxy.v1.config.RouteRule
 import io.buoyant.namer.RichActivity
+import istio.proxy.v1.config.RouteRule
 
 class RouteManager(api: IstioPilotClient, pollInterval: Duration) {
-  private[this] val states: Var[Activity.State[Map[String, RouteRule]]] = api.watchRouteRules(pollInterval).run.map {
-    case Activity.Ok(rsp) =>
-      log.trace("istio route-rules received")
-      Activity.Ok(mkRouteMap(rsp))
-    case Activity.Pending =>
-      log.trace("istio route manager is pending")
-      Activity.Pending
-    case Activity.Failed(e) =>
-      log.error(e, "failed to update istio routing rules")
-      Activity.Failed(e)
-  }
+  private[this] val states: Activity[Map[String, RouteRule]] = api.watchRouteRules(pollInterval).map(mkRouteMap)
 
   private[this] def mkRouteMap(routeList: Seq[RouteRuleConfig]): Map[String, RouteRule] =
-    routeList.flatMap {
-      case RouteRuleConfig(typ, Some(name), Some(spec)) => Some(name -> spec)
-      case _ => None
-    }.toMap[String, RouteRule]
+    routeList.collect {
+      case RouteRuleConfig(typ, Some(name), Some(spec)) => name -> spec
+    }.toMap
 
   private[this] lazy val routeRules: Activity[Map[String, RouteRule]] = {
-    val act = Activity(states)
-    val _ = act.states.respond(_ => ()) // register a listener forever to keep the Activity open
-    act
+    val _ = states.states.respond(_ => ()) // register a listener forever to keep the Activity open
+    states
   }
 
   def getRules(): Future[Map[String, RouteRule]] = routeRules.toFuture
 }
+
+object RouteManager {
+  private[this] val managers = Var(Map.empty[(String, Int), RouteManager])
+
+  def getManagerFor(host: String, port: Int): RouteManager = {
```

Comment:
TIOLI: could use com.twitter.util.Memoize instead of a Var.  Or even just a mutable Map.